python-backend/main.py

```diff
@@ -31,42 +33,74 @@
                    allow_headers=["*"])
 
 
+def parse(data: Input) -> Tuple[Mul, Add, Symbol]:
+    """
+    Process user inputs into math expressions
+    :param data: User form inputs
+    :return: tuple including p/q, simplified q and the symbol/variable used in these expressions
+    """
+    x = Symbol(data.symbol, real=True)
+    p = sympify(convert(data.p), {data.symbol: x})
+    q = sympify(convert(data.q), {data.symbol: x})
+    simple_q = simplify(q)
+    expression = p / simple_q
+    simple = simplify(expression)
+    return simple, simple_q, x
+
+
 @app.post("/analyze")
 async def analyze(request: Request):
     """
+    Take user form inputs and parse them into mathematical expressions, then assess them in various ways and return
+    chart coordinates to be rendered by the frontend
     :param request: HTTP request
     :return: HTTP response indicating success of parsing inputs with a 200 or a 500 to indicate failure parsing inputs
     """
     # parse posted body as Input
 
-    mpmath.mp.dps = PRECISION
-
     try:
         data = Input(**(await request.json()))
-        x = Symbol(data.symbol, real=True)
-        p = sympify(convert(data.p), {data.symbol: x})
-        q = sympify(convert(data.q), {data.symbol: x})
-        simple_q = simplify(q)
-        expression = p / simple_q
-        simple = simplify(expression)
+        (expression, denominator, symbol) = parse(data)
 
-        q_limit = sympy.limit(simple_q, x, oo)
-        limit = sympy.limit(simple, x, oo)
+        limit = sympy.limit(expression, symbol, oo)
 
         body = {
-            "wolfram_limit": WolframClient.limit(str(simple)),
-            "limit": json.dumps(float(limit)),
-            "denominator_limit": json.dumps(float(q_limit)),
-            "log_error": json.dumps(error_coordinates(simple, x, limit)),
-            "delta": json.dumps(delta_coordinates(simple, simple_q, x, limit)),
-            "computed_value": json.dumps(float(limit))  # @TODO: replace with actual computation to i
+            "expression": json.dumps(str(expression)),
+            "limit": json.dumps("Infinity" if type(limit) is Infinity else str(limit)),
+            "log_error": json.dumps(error_coordinates(expression, symbol, limit)),
+            "delta": json.dumps(delta_coordinates(expression, denominator, symbol, limit)),
+            "computed_value": 0  # @TODO: replace with actual computation to i
         }
-
+        logger.debug(f"Response: {body}")
         response = JSONResponse(content=body)
         response.set_cookie(key="trm", value=str(uuid.uuid4()))
         return response
 
     except Exception as e:
         logger.warning(e)
-        response = JSONResponse(status_code=500, content={"error": "Failed to parse p / q"})
+        response = JSONResponse(status_code=500, content={"error": "Failed to parse inputs"})
```

Comment:
I'm assuming that this code won't be invoked if the user provides bad input, but I'm not seeing where this is handled above. If bad input is handled elsewhere, consider adding a comment here. If bad input isn't handled elsewhere, consider differentiating between bad input (4XX) and bad implementation (5XX).

Reply:
The error text is outdated. Parsing is not the problem anymore so I have changed the message and left it at 500.